crucible-mir/src/Mir/TransCustom.hs

```diff
@@ -624,6 +630,54 @@ null_ptr_impl what substs = case substs of
         _  -> mirFail $ "expected no arguments for " ++ what ++ ", received: " ++ show ops
     _ -> Nothing
 
+-- | Experimentally, we've observed that rustc seems not to generate proper drop
+-- glue for @&dyn Trait@ drops. We get around this by overriding
+-- @core::ptr::drop_in_place@ (when instantiated at @dyn@ types) to fetch the
+-- appropriate drop method from the trait object's vtable. @mir-json@ will have
+-- included this method.
+drop_in_place_dyn :: (ExplodedDefId, CustomRHS)
+drop_in_place_dyn =
+    ( ["core", "ptr", "drop_in_place"],
+      \case
+        Substs [TyDynamic traitName] ->
+            Just $ CustomOp $ \argTys [op@(MirExp selfTy selfExpr)] -> do
+                col <- use $ cs . collection
+                let argTys = Ctx.empty
+                let argExprs = Ctx.empty
+                let retTy = C.UnitRepr
+
+                -- We expect @mir-json@ to have placed this trait object's drop
+                -- method at index 0, unless the trait object lacks a principal
+                -- trait. Check here whether the trait has any methods, to emit
+                -- a more relevant error message than `doVirtCall` would emit on
+                -- its own.
```

Comment:
Would be good to note that we consider the current mir-json behavior for `dyn Send` to be a bug/limitation - mir-json should emit a vtable with only a drop and no other methods, but that would take some work, so we haven't done it yet.

Reply:
Done in 5d54bef.